src/chapters/angular-lsn2/studio.rst

```diff
@@ -0,0 +1,265 @@
+
+.. sourcecode:: TypeScript
+   :linenos:
+
+   remove(member: object) {
+     let index = this.crew.indexOf(member);
+     this.crew.splice(index, 1);
+   }
+
+Edit Crew Members
+^^^^^^^^^^^^^^^^^
+Finally we are going to allow the user to edit crew members that have already been added.
+If the crew member name is clicked, then the crew member name should be replaced with a text
+input and a save button. This means that the crew member name can be displayed in display state
+or in edit state. Only one crew member can be edited at once.
+
+.. figure:: figures/edit-crew-name.gif
+       :alt: Animated gif of crew member name being clicked, edited, and then saved.
+
+       Example of crew member name being edited.
+
+We need to add a click event to the member name. To do that, put ``{{member.name}}`` inside of a
+``<span>`` that has a ``(click)`` handler. Make your ``<li>`` in ``src/app/crew/crew.component.html``
+look like the below code.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <li *ngFor="let member of crew">
+      <span (click)="edit(member)" class="editable-text">{{member.name}}</span>
+      <button (click)="remove(member)">remove</button>
+   </li>
+
+Add this property to the crew component. ``memberBeingEdited`` represents the crew
+member that is currently being edited.
+
+.. sourcecode:: TypeScript
+
+   memberBeingEdited: object = null;
+
+Next we need to add the below ``edit`` function to the crew component file ``src/app/crew/crew.component.ts``.
+This function will set a ``memberBeingEdited`` variable to be equal to the crew member that was clicked.
+
+.. sourcecode:: TypeScript
+
+   edit(member: object) {
+      this.memberBeingEdited = member;
+   }
+
+Now we need to add an ``*ngIf`` that will show the two versions of the member, the display state or the edit state.
+In the edit state an input box with a save button will appear, but for now the input and save won't have any functionality.
+Make your ``src/app/crew/crew.component.html`` file look like the below code.
+
+.. sourcecode:: html+ng2
+   :linenos:
+
+   <h3>Crew</h3>
+   <ul>
+      <li *ngFor="let member of crew">
+
+         <span *ngIf="memberBeingEdited !== member; else elseBlock">
+            <!-- display state of member -->
+            <span (click)="edit(member)" class="editable-text">{{member.name}}</span>
+            <span *ngIf="member.firstMission">
+               - 1st
+            </span>
+            <button (click)="remove(member)">remove</button>
+         </span>
+
+         <ng-template #elseBlock>
+            <!-- edit state of member -->
+            <input />
+            <button>save</button>
+         </ng-template>
+
+      </li>
+   </ul>
+   <input #name type="text"/>
+   <label>First mission<input #firstMission type="checkbox"/></label>
+   <button (click)="add(name.value, firstMission.checked)">Add</button>
+
+Finally we are going to make the edit state of the member update the member name when save is clicked.
```

Comment:
Word removal: "Finally we are going to make the edit state update the member name when "Save" is clicked."